tests/admin.test.ts

```diff
@@ -0,0 +1,83 @@
+import { ActivityScope, ActivityType, UserAccessType } from '../types';
+import { ControllerFactory } from './controllers';
+import { DatabaseConnection, EventFactory, UserFactory, PortalState } from './data';
+
+beforeAll(async () => {
+  await DatabaseConnection.connect();
+});
+
+beforeEach(async () => {
+  await DatabaseConnection.clear();
+});
+
+afterAll(async () => {
+  await DatabaseConnection.clear();
+  await DatabaseConnection.close();
+});
+
+describe('retroactive attendance submission', () => {
+  test('logs activity, attendance, and points for users who have not attended', async () => {
+    const users = UserFactory.create(3);
+    const emails = users.map((user) => user.email);
+    const [proxyUser] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const [event] = EventFactory.create(1);
+
+    await new PortalState()
+      .createUsers([...users, proxyUser])
+      .createEvents([event])
+      .write();
+
+    const adminController = await ControllerFactory.admin();
+    const userController = await ControllerFactory.user();
+    const attendanceController = await ControllerFactory.attendance();
+
+    await adminController.submitAttendanceForUsers({ users: emails, event: event.uuid }, proxyUser);
+
+    await Promise.all(users.map(async (user) => {
```

Comment:
Just tried a few ways of iterating over and it seems like `Promise.all` is the only one that works currently:
* `for/of` throws a linter error to not use `for/of` (no-restricted-syntax)
* just `for` throws linter errors for no-await-in-loop
* `users.forEach(async (user) => {...});` fails the test cases because for some reason a `NotFoundError` is being thrown saying that a user is not found. Currently have no clue why this is happening - I tried reading the users table from `conn.manager.find` and all the data is there.
* `Promise.all` works just fine, with the one downside that the order is non-deterministic

Reply:
We could disable `no-await-in-loop` for tests in the linter config (set a pattern match for file names `*.test.ts` or something similar). I personally find the `Promise.all` pretty weird and prefer looping, but it's your call.

Reply:
Yeah agreed `Promise.all` is somewhat sketch in this situation since it's just a hacky way of looping through elements to pass the linter